jekyll/_cci2/bundling-config.md

```diff
@@ -0,0 +1,905 @@
+        default: "I'm a default value"
+        type: string
+
+    steps:
+      - run: echo '<< parameters.foo >>'
+```
+
+In the above example a value for the parameter `foo` can be passed when invoking the command. For instance, passing the parameter `foo` would look something like:
+
+```yaml
+jobs:
+  myjob:
+    docker:
+      - image: circleci/ruby:2.4-node
+    steps:
+      - bar:
+          foo: "I'm the passed value"
+```
+
+### Parameter Syntax
+{:.no_toc}
+
+A parameter can have the following keys as immediate children:
+
+| Key Name    | Description                                                                                   | Default value |
+|-------------|-----------------------------------------------------------------------------------------------|---------------|
+| description | Optional. Used to generate documentation for your orb.                                        | N/A           |
+| type        | Required. Currently "string", "boolean", "enum" (takes extra keys), and "steps" are supported                           | N/A           |
+| default     | The default value for the parameter. If not present, the parameter is implied to be required. | N/A           |
+
+### Parameter Types
+{:.no_toc}
+
+This section describes the types of parameters and their usage.
+
+### Parameter Scope
+{:.no_toc}
+
+Parameters are in-scope only within the job or command that defined them. If you want a job or command to pass its parameters to a command it invokes, they must be passed explicitly. Command, job, executor, and parameter names can only contain lowercase letters a-z, digits, and _ and -, and must start with a letter.
+
+
+```yaml
+version: 2.1
+
+jobs:
+  sayhello:
+    parameters:
+      saywhat:
+        description: "To whom shall we say hello?"
+        default: "World"
+        type: string
+    machine: true
+    steps:
+      - say:
+          # Since the command "say" doesn't define a default
+          # value for the "saywhat" parameter, it must be
+          # passed in manually
+          saywhat: << parameters.saywhat >>
+
+commands:
+  say:
+    parameters:
+      saywhat:
+        type: string
+    steps:
+      - run: echo "<< parameters.saywhat >>"
+
+workflows:
+  build:
+    jobs:
+      - sayhello:
+          saywhat: Everyone
+```
+
+
+#### String
+{:.no_toc}
+
+Basic string parameters are described below:
+
+```yaml
+commands:
+  copy-markdown:
+    parameters:
+      destination:
+        description: destination directory
+        type: string
+        default: docs
+    steps:
+      - cp *.md << destination >>
+```
+
+Strings should be quoted if they would otherwise represent another type (such as boolean or number) or if they contain characters that have special meaning in yaml, particularly for the colon character. In all other instances, quotes are optional. Empty strings are treated as a falsy value in evaluation of `when` clauses, and all other strings are treated as truthy.
```

Comment:
`and all other strings are treated as truthy.` seems misleading. I think this is true, but some unquoted strings can be coerced into booleans, like `yes`, `on`, and `off`. I think this phrase should be removed.

Reply:
We don't allow unquoted booleans as the values of a `string` type (which we should probably mention) -- in `when` and `unless` we count all non-empty strings as truthy, including the string "false" and "no" and "off"

Reply:
Ah that's an interesting subtlety. In that case, my original comment in this thread can be disregarded. I agree with Nathan's point: using an unquoted string value that YAML interprets as a boolean will result in a type error.

Reply:
thanks, I added this last sentence to the paragraph.